adapter/internal/discovery/xds/rate_limiter_cache_test.go

```diff
@@ -577,3 +578,86 @@ func getDummyAPISwagger(apiID, level, apiPolicy, res1GetPolicy, res1PostPolicy,
 	mgwSwagger.VHost = "vhost1"
 	return mgwSwagger
 }
+func TestAddSubscriptionLevelRateLimitPolicy(t *testing.T) {
+	policyList := &types.SubscriptionPolicyList{
+		List: []types.SubscriptionPolicy{
+			{
+				Name: "Policy1",
+				DefaultLimit: &types.SubscriptionDefaultLimit{
+					QuotaType: "requestCount",
+					RequestCount: &types.SubscriptionRequestCount{
+						RequestCount: 100,
+						TimeUnit:     "sec",
+					},
+				},
+				Organization: "org1",
+			},
+			{
+				Name: "Policy2",
+				DefaultLimit: &types.SubscriptionDefaultLimit{
+					QuotaType: "requestCount",
+					RequestCount: &types.SubscriptionRequestCount{
+						RequestCount: 200,
+						TimeUnit:     "min",
+					},
+				},
+				Organization: "org1",
+			},
+			{
+				Name: "Unauthenticated",
+				DefaultLimit: &types.SubscriptionDefaultLimit{
+					QuotaType: "requestCount",
+					RequestCount: &types.SubscriptionRequestCount{
+						RequestCount: 300,
+						TimeUnit:     "hours",
+					},
+				},
+				Organization: "org1",
+			},
+			{
+				Name: "AsyncPolicy1",
+				DefaultLimit: &types.SubscriptionDefaultLimit{
+					QuotaType: "eventCount",
+					RequestCount: &types.SubscriptionRequestCount{
+						RequestCount: 300,
+						TimeUnit:     "hours",
+					},
+				},
+				Organization: "org1",
+			},
+		},
+	}
+
+	// Initialize rlsPolicyCache.metadataBasedPolicies
+	rlsPolicyCache.metadataBasedPolicies = make(map[string]map[string]map[string]*rls_config.RateLimitDescriptor)
```

Comment:
This is initialized inside `AddSubscriptionLevelRateLimitPolicy` neh? so we can remove this line